benchmarks/benchmarks/bench_indexing.py

```diff
@@ -84,6 +84,27 @@ def time_assign_cast(self, ndim):
             arr[indx] = val
 
 
+class BooleanAssignmentOrder(Benchmark):
+    params = ['C', 'F']
+    param_names = ['order']
+
+    def setup(self, order):
+        shape = (64, 64, 64)
+        self.base = np.zeros(shape, dtype=np.uint32, order=order)
+        # emulate gh-30156: boolean assignment into a sliced Fortran/C view
+        self.view = self.base[1:-1, 1:-1, 1:-1]
+        mask = np.random.RandomState(0).rand(*self.view.shape) > 0.5
+        if order == 'F':
+            mask = np.asfortranarray(mask)
+        else:
+            mask = np.ascontiguousarray(mask)
+        self.mask = mask
```

Comment:
```suggestion
        self.mask = mask.copy(order)
```
or similar also works, not that it matters.